nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -590,9 +595,10 @@ private void handleEnhancedForOverKeySet(
                 .getMapGetIteratorContentsAccessPath((LocalVariableNode) receiver);
         if (mapGetPath != null) {
           // put sanity check here to minimize perf impact
-          if (!isCallToMethod(methodInv, "java.util.Iterator", "next")) {
+          if (!isCallToMethod(methodInv, ITERATOR_TYPE_SUPPLIER, "next")) {
```

Comment:
This is a key change; we allow `next()` to come from a subtype of `Iterator` (didn't see a need for this yet but I could easily see it happening)